nibabel/streamlines/trk.py

```diff
@@ -0,0 +1,713 @@
+                msg = "Nb. of scalars differs from one point to another!"
+                raise DataError(msg)
+
+            if nb_properties_per_streamline != int(nb_properties_per_streamline):
+                msg = ("Nb. of properties differs from one streamline to"
+                       " another!")
+                raise DataError(msg)
+
+            header[Field.NB_STREAMLINES] = nb_streamlines
+            header[Field.NB_SCALARS_PER_POINT] = nb_scalars_per_point
+            header[Field.NB_PROPERTIES_PER_STREAMLINE] = nb_properties_per_streamline
+
+            # Overwrite header with updated one.
+            f.seek(beginning, os.SEEK_SET)
+            f.write(header.tostring())
+
+    @staticmethod
+    def _read_header(fileobj):
+        """ Reads a TRK header from a file.
+
+        Parameters
+        ----------
+        fileobj : string or file-like object
+            If string, a filename; otherwise an open file-like object
+            pointing to TRK file (and ready to read from the beginning
+            of the TRK header). Note that calling this function
+            does not change the file position.
+
+        Returns
+        -------
+        header : dict
+            Metadata associated to this tractogram file.
+        """
+        with Opener(fileobj) as f:
+            start_position = f.tell()
+
+            # Read the header in one block.
+            header_str = f.read(header_2_dtype.itemsize)
+            header_rec = np.fromstring(string=header_str, dtype=header_2_dtype)
+
+            # Check endianness
+            endianness = native_code
+            if header_rec['hdr_size'] != TrkFile.HEADER_SIZE:
+                endianness = swapped_code
+
+                # Swap byte order
+                header_rec = header_rec.newbyteorder()
+                if header_rec['hdr_size'] != TrkFile.HEADER_SIZE:
+                    msg = "Invalid hdr_size: {0} instead of {1}"
+                    raise HeaderError(msg.format(header_rec['hdr_size'],
+                                                 TrkFile.HEADER_SIZE))
+
+            if header_rec['version'] == 1:
+                header_rec = np.fromstring(string=header_str,
+                                           dtype=header_1_dtype)
+            elif header_rec['version'] == 2:
+                pass  # Nothing more to do.
+            else:
+                raise HeaderError('NiBabel only supports versions 1 and 2.')
+
+            # Convert the first record of `header_rec` into a dictionnary
+            header = dict(zip(header_rec.dtype.names, header_rec[0]))
+            header[Field.ENDIANNESS] = endianness
+
+            # If vox_to_ras[3][3] is 0, it means the matrix is not recorded.
+            if header[Field.VOXEL_TO_RASMM][3][3] == 0:
+                header[Field.VOXEL_TO_RASMM] = np.eye(4, dtype=np.float32)
+                warnings.warn(("Field 'vox_to_ras' in the TRK's header was"
```

Comment:
Good question. I followed what was already done in `nibabel.trackvis` (see https://github.com/nipy/nibabel/blob/master/nibabel/trackvis.py#L52).